src/Extension/Rosie/CodigaConfigFileUtil.cs

```diff
@@ -0,0 +1,111 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+using Solution = EnvDTE.Solution;
+
+namespace Extension.Rosie
+{
+    /// <summary>
+    /// Utility for retrieving information about and from the codiga.yml config file in the Solution.
+    /// </summary>
+    public static class CodigaConfigFileUtil
+    {
+        /// <summary>
+        /// The naming convention makes it possible to parse the rulesets property as lowercase 'rulesets',
+        /// and not as 'Rulesets' (what the C# class property is called in <see cref="CodigaCodeAnalysisConfig"/>.  
+        /// </summary>
+        private static readonly IDeserializer ConfigDeserializer = new DeserializerBuilder()
+            .WithNamingConvention(LowerCaseNamingConvention.Instance)
+            .Build();
+        private static readonly Regex CodigaRulesetNamePattern = new Regex("^[a-z0-9][a-z0-9-]{4,}$");
```

Comment:
`[a-z0-9-]{5,32}$`

5 is the min.
32 is the max.

This is closer to the backend regex.

Reply:
After chatting with @picimako, this is more accurate.

`^[a-z0-9][a-z0-9-]{5, 32}$`